controller/api/destination/watcher/endpoints_watcher.go

```diff
@@ -129,6 +129,12 @@ func (ew *EndpointsWatcher) Subscribe(authority string, listener EndpointUpdateL
 	if err != nil {
 		return err
 	}
+
+	svc, _ := ew.k8sAPI.Svc().Lister().Services(id.Namespace).Get(id.Name)
+	if svc != nil && svc.Spec.Type == corev1.ServiceTypeExternalName {
+		return invalidService(authority)
```

Comment:
When I set the `destination` container's log level to `debug`, the logs are completely saturated with the `Failed to subscribe` error messages as seen below. I like to think this won't be a problem, as users would only turn on `debug` for temporary debugging purposes, but thought I will bring it to up anyway.

```
time="2019-07-23T03:25:15Z" level=error msg="Failed to subscribe to my-external-nginx.default.svc.cluster.local:80: Invalid k8s service my-external-nginx.default.svc.cluster.local:80" addr=":8086" component=server remote="127.0.0.1:41306"
time="2019-07-23T03:25:15Z" level=debug msg="Get my-external-nginx.default.svc.cluster.local:80" addr=":8086" component=server remote="127.0.0.1:41306"
time="2019-07-23T03:25:15Z" level=error msg="Failed to subscribe to my-external-nginx.default.svc.cluster.local:80: Invalid k8s service my-external-nginx.default.svc.cluster.local:80" addr=":8086" component=server remote="127.0.0.1:41306"
time="2019-07-23T03:25:15Z" level=debug msg="Get my-external-nginx.default.svc.cluster.local:80" addr=":8086" component=server remote="127.0.0.1:41306"
time="2019-07-23T03:25:15Z" level=error msg="Failed to subscribe to my-external-nginx.default.svc.cluster.local:80: Invalid k8s service my-external-nginx.default.svc.cluster.local:80" addr=":8086" component=server remote="127.0.0.1:41306"
time="2019-07-23T03:25:15Z" level=debug msg="Get my-external-nginx.default.svc.cluster.local:80" addr=":8086" component=server remote="127.0.0.1:41306"
time="2019-07-23T03:25:15Z" level=error msg="Failed to subscribe to my-external-nginx.default.svc.cluster.local:80: Invalid k8s service my-external-nginx.default.svc.cluster.local:80" addr=":8086" component=server remote="127.0.0.1:41306"
time="2019-07-23T03:25:15Z" level=debug msg="Get my-external-nginx.default.svc.cluster.local:80" addr=":8086" component=server remote="127.0.0.1:41306"
time="2019-07-23T03:25:15Z" level=error msg="Failed to subscribe to my-external-nginx.default.svc.cluster.local:80: Invalid k8s service my-external-nginx.default.svc.cluster.local:80" addr=":8086" component=server remote="127.0.0.1:41306"
time="2019-07-23T03:25:15Z" level=debug msg="Get my-external-nginx.default.svc.cluster.local:80" addr=":8086" component=server remote="127.0.0.1:41306"
time="2019-07-23T03:25:15Z" level=error msg="Failed to subscribe to my-external-nginx.default.svc.cluster.local:80: Invalid k8s service my-external-nginx.default.svc.cluster.local:80" addr=":8086" component=server remote="127.0.0.1:41306"
time="2019-07-23T03:25:15Z" level=debug msg="Get my-external-nginx.default.svc.cluster.local:80" addr=":8086" component=server remote="127.0.0.1:41306"
time="2019-07-23T03:25:15Z" level=error msg="Failed to subscribe to my-external-nginx.default.svc.cluster.local:80: Invalid k8s service my-external-nginx.default.svc.cluster.local:80" addr=":8086" component=server remote="127.0.0.1:41306"
time="2019-07-23T03:25:15Z" level=debug msg="Get my-external-nginx.default.svc.cluster.local:80" addr=":8086" component=server remote="127.0.0.1:41306"
time="2019-07-23T03:25:15Z" level=error msg="Failed to subscribe to my-external-nginx.default.svc.cluster.local:80: Invalid k8s service my-external-nginx.default.svc.cluster.local:80" addr=":8086" component=server remote="127.0.0.1:41306"
time="2019-07-23T03:25:15Z" level=debug msg="Get my-external-nginx.default.svc.cluster.local:80" addr=":8086" component=server remote="127.0.0.1:41306"
time="2019-07-23T03:25:15Z" level=error msg="Failed to subscribe to my-external-nginx.default.svc.cluster.local:80: Invalid k8s service my-external-nginx.default.svc.cluster.local:80" addr=":8086" component=server remote="127.0.0.1:41306"
time="2019-07-23T03:25:15Z" level=debug msg="Get my-external-nginx.default.svc.cluster.local:80" addr=":8086" component=server remote="127.0.0.1:41306"
time="2019-07-23T03:25:15Z" level=error msg="Failed to subscribe to my-external-nginx.default.svc.cluster.local:80: Invalid k8s service my-external-nginx.default.svc.cluster.local:80" addr=":8086" component=server remote="127.0.0.1:41306"
time="2019-07-23T03:25:15Z" level=debug msg="Get my-external-nginx.default.svc.cluster.local:80" addr=":8086" component=server remote="127.0.0.1:41306"
time="2019-07-23T03:25:15Z" level=error msg="Failed to subscribe to my-external-nginx.default.svc.cluster.local:80: Invalid k8s service my-external-nginx.default.svc.cluster.local:80" addr=":8086" component=server remote="127.0.0.1:41306"
...
```

Reply:
Looks like half of those are logged at error level, which is problematic...